rmf_task_ros2/src/rmf_task_ros2/Dispatcher.cpp

```diff
@@ -474,9 +479,19 @@ class Dispatcher::Implementation
       }
 
       const auto& task_request_json = msg_json["request"];
-      const std::string task_id =
+      std::string task_id =
         task_request_json["category"].get<std::string>()
-        + ".dispatch-" + std::to_string(task_counter++);
+        + ".dispatch-";
+
+      if (use_timestamp_for_task_id)
+      {
+        task_id += std::to_string(
+          static_cast<int>(node->get_clock()->now().nanoseconds()/1e9));
```

Comment:
Wondering if we should leave the timestamp in nanoseconds and not divide by 1e9 to get seconds? If somehow two or more tasks get processed within the same second, they might have the same task_id?

Reply:
Yea, that is valid concern that i thought about too. The only concern (probably insignificant) is having nanosec as timestamp is way too long. This will result in a long `task_id` that might look quite ugly on the dashboard? Example shown here:

![Screenshot from 2022-08-19 15-40-39](https://user-images.githubusercontent.com/32189404/185569084-047b36b3-6afa-442c-8e14-ac817fb4a52a.png)

Let me nego abit, how about milisec? Since  `unix_millis` time is regularly used when defining our rmf time field.

Reply:
Okay let's go with millis for now.
I do this this feature will be redundant once we implement a proper backup and restore functionality for this node. That way when the node is restored after a crash, it will know the last issued task_id.